include/bluetooth/conn.h

```diff
@@ -277,6 +277,50 @@ int bt_conn_le_param_update(struct bt_conn *conn,
  */
 int bt_conn_disconnect(struct bt_conn *conn, u8_t reason);
 
+enum {
+	/** Convenience value when no options are specified. */
+	BT_LE_CONN_OPT_NONE = 0,
+
+	/* Enable scan on coded PHY, accept starting connection on CODED */
+	BT_LE_CONN_OPT_CODED = BIT(0),
+
+	/* Accept starting connection on 2M */
+	BT_LE_CONN_OPT_2M = BIT(1),
+
+	/* Disable scan on 1M PHY, disable starting connection on 1M. */
+	BT_LE_CONN_OPT_NO_1M = BIT(2),
+};
+
+struct bt_conn_scan_param {
+
+	/** Bit-field of scanning options. */
+	u32_t options;
+
+	/** Scan interval (N * 0.625 ms) */
+	u16_t interval;
+
+	/** Scan window (N * 0.625 ms) */
+	u16_t window;
+
+	/** Scan interval coded (long range) (N * 0.625 MS)
+	 *  Set zero to use same as 1M scanning
+	 */
+	u16_t interval_coded;
+
+	/** Scan window coded (long range) (N * 0.625 MS)
+	 *  Set zero to use same as 1M scanning
+	 */
+	u16_t window_coded;
+};
+
+/* @brief Set scan parameters used when initiating an LE Connection.
+ *
+ *  @param param Scan parameters
+ *
+ *  @return Zero on success or (negative) error code on failure.
+ */
+int bt_conn_set_scan_params(const struct bt_conn_scan_param *param);
```

Comment:
I did it as a way of keeping backwards compatebility yes. And the host hase to hold on to these parameters since they are included in every call to LE Create Connection command.

Creating a new connect function is an option. But I though it would be more confusing than not. Removing the _le suffix would kind of seem like it could by used for both LE and BR/EDR.
Adding `bt_conn_create_master_le`, since we have slave version? (not the best name in my opinion).

I'm open for suggestions.

Reply:
I agree with the proposal [here](https://github.com/zephyrproject-rtos/zephyr/pull/22013#discussion_r375748828). I think we should start deprecating the `_le_` infix and use names like `bt_conn_create()`

Reply:
Unless @jhedberg disagrees, deprecating the two existing functions:
- bt_conn_create_le()
- bt_conn_create_auto_le()

and removing the `_le()` suffix for the new versions is the way to go.

Reply:
Summary from offline discussion(s):
Rename to `bt_conn_le_create` and `bt_conn_le_create_auto`, switch to returning int, and having conn object as out parameter.